sklearnex/preview/ensemble/forest.py

```diff
@@ -1051,6 +1186,8 @@ def _onedal_gpu_supported(self, method_name, *data):
                 return False
             elif self.oob_score:
                 return False
+            elif sklearn_check_version("1.4") and self.monotonic_cst is not None:
+                return False
             else:
                 return True

```

Comment:
This `if...elif...else` statement is just horrifying

Reply:
I can't disagree, but it's out of PR scope.